### PR TITLE
tests: subsys: logging: move log_syst to new ztest API

### PR DESCRIPTION
**Pull request by 1xlibiao**
Move test subsys/logging/log_syst to use new ztest API.

Signed-off-by: li biao <biao1x.li@intel.com>

### CODE REVIEW

**Comment by 1xlibiao** [comment]
> Why not use ZTEST() for every test?

 I have tried,but it reported errors

**Comment by 1xlibiao** [comment]
> Why not use ZTEST() for every test?

There is also a .c file below the src file, they are related, so I used this method

**Comment by dcpleung** [comment]
What was the error?

**Comment by 1xlibiao** [comment]
> What was the error?

If you do not modify in this way, it will display an error that is not in the same order as the initial code execution

**Comment by 1xlibiao** [comment]
Test strategy as below:

**Verified Platforms:**

- qemu_x86
- qemu_cortex_m0
- qemu_cortex_m3
- frdm_k64f
- qemu_riscv32_smp

Shuffled tests are failed

If I use ZTEST() for every test,an error will be reported that is not in the original execution order,so I tried to encapsulate other functions with a ZTEST().

File: tests/subsys/logging/log_syst/src/main.c [hunk 1]
```diff
@@ -129,13 +129,12 @@ void test_log_syst_float_data(void)
...
-		ztest_unit_test(test_log_syst_format_table_selection),
-		ztest_unit_test(test_log_syst_data),
-		ztest_unit_test(test_log_syst_float_data),
-		ztest_unit_test(test_log_syst_data_multiple_args)
-		);
-	ztest_run_test_suite(test_log_syst);
+	test_log_syst_format_table_selection();
+	test_log_syst_data();
+	test_log_syst_float_data();
+	test_log_syst_data_multiple_args();
```
Comment (enjiamai) [inline tests/subsys/logging/log_syst/src/main.c:137]: Please fixed it, do not put them together unless you understand it and you have to, thanks.

**Review by enjiamai — CHANGES_REQUESTED** [review]

File: tests/subsys/logging/log_syst/src/main.c (hunk 1 above)
Comment (1xlibiao) [reply tests/subsys/logging/log_syst/src/main.c:137]: > Please fixed it, do not put them together unless you understand it and you have to, thanks.

ok,I will.thanks

**Pull request merged**